libeos-payg/efi.c

```diff
@@ -559,6 +559,26 @@ eospayg_efi_secureboot_active (void)
   return !!content[0];
 }
 
+/* eospayg_efi_setupmode_active:
+ *
+ * Check if the system was booted with SetupMode
+ *
+ * Returns: %TRUE if booted with SetupMode, %FALSE otherwise
+ */
+gboolean
+eospayg_efi_setupmode_active (void)
+{
+  g_autofree char *tname = full_efi_name (GLOBAL_VARIABLE_GUID, "SetupMode");
+  g_autofree unsigned char *content = NULL;
+  int size;
+
+  content = efivarfs_read (tname, &size, NULL);
```

Comment:
As discussed you need to either handle `test_mode` specially, or call `efi->read` which itself handles `test_mode` specially (assuming FALSE is the correct value for the test suite by default), or equivalently add `eospayg_efi_var_read_fullname` and call that.